SPDX-FileCopyrightText: 2026 Kalle Fagerberg

SPDX-License-Identifier: CC0-1.0

diff --git a/bin/klock.kubectl-klock.completion.bash b/bin/klock.kubectl-klock.completion.bash
index 0a73fd2..ecedbc2 100755
--- a/bin/klock.kubectl-klock.completion.bash
+++ b/bin/klock.kubectl-klock.completion.bash
@@ -21,14 +21,24 @@ __klock.kubectl-klock_get_completion_results() {
     local requestComp lastParam lastChar args
 
     # Prepare the command to request completions for the program.
     # Calling ${words[0]} instead of directly klock.kubectl-klock allows handling aliases
     args=("${words[@]:1}")
     requestComp="${words[0]} __complete ${args[*]}"
 
+    ##########################
+    ## Custom fix for snap  ##
+    ##########################
+    if [[ -x "$SNAP/kubectl-klock" ]]; then
+        requestComp="$SNAP/kubectl-klock __complete ${args[*]}"
+    fi
+    ##########################
+    ##  End of custom fix   ##
+    ##########################
+
     lastParam=${words[$((${#words[@]}-1))]}
     lastChar=${lastParam:$((${#lastParam}-1)):1}
     __klock.kubectl-klock_debug "lastParam ${lastParam}, lastChar ${lastChar}"
 
     if [[ -z ${cur} && ${lastChar} != = ]]; then
         # If the last parameter is complete (there is a space following it)
         # We add an extra empty parameter so we can indicate this to the go method.
